resources/fetchersManager/factory.go

```diff
@@ -58,28 +58,6 @@ func (fa *factories) CreateFetcher(log *logp.Logger, name string, c *agentconfig
 	return factory.Create(log, c, ch)
 }
 
-func (fa *factories) RegisterFetchers(log *logp.Logger, registry FetchersRegistry, cfg config.Config, ch chan fetching.ResourceInfo) error {
```

Comment:
wasn't removed, but moved to a place that makes more sense to me. We have the factory that should handle the factories and we have the fetcher registry that handles registration fetchers, other implementation details shouldn't change these two facts. And regarding newcomers that will find this change complex, from my own experience here, the old design really made it hard for me.

> I think that the idea was that the factory will have one function that will register all of the fetchers.

exactly, why should the factory care about the registry job?

> The author of the original code tries to encapsulate all the registration in a single line of code, so it will be easier to use the factory.

what does that mean? a single line of code? easier to use the factory? why should we use the factory to register the fetchers in the first place?